### PR TITLE
docs(plan): Phase 0 decisions for the 2026-08-12 improvement notes

### PR DESCRIPTION
**Pull request by CodySwannGT**
## What this is

Phase 0 of `plans/improvement-notes-implementation.md`: the reviewed improvement
notes, the phased implementation plan, and the four decision records that unblock
Phase 2.

Work-Item: CodySwannGT/lisa#2423

## Why the notes were revised before planning

Every "current state" claim in the notes was verified against the codebase. Three were
overstated and are corrected in this PR:

- **Secrets (item 3):** the contract claims provider notes must be "well-formed", but
  nothing checks format — `doctor-secrets.mjs` only tests for a non-empty note and
  reports it as a *warn*. The `verify` surface is the skill's own operation, not the
  `lisa-verify` flow.
- **Ready role (item 9):** `exploratory-qa` deliberately files findings with
  `ready=false`, so "automated loops already file build-ready" was wrong. Separately,
  omitted `build_ready` means *not ready* on JIRA but *ready* on GitHub and Linear.
- **BDD (item 18):** "every obligation has a mapping or dated waiver" is a
  `lisa-verify` flow obligation, not a gate rule — the gate's enforced defect set is
  only `empty-contract` / `floor-missing` / `floor-regression`. `exclusions` is
  documented and templated but read by zero gate code.

## Fleet audit

A survey of tunnlai, geminisportsai, and propswap (origin default branches) supplied
the evidence behind work units K–N, including bugs in Lisa that downstream repos have
been patching around: guard scripts that never reach installed repos on a version
bump, a learnings ledger scaffolded into the auto-loaded rules tree (19 captured
entries silently lost to a merge), a destructive ledger trim, and a Linear `ready`
lane mapped to `Todo`.

One audited claim has already been **refuted** during Phase 1 and is noted for the
downstream cleanup: the `skip_jobs` substring-matching bug was fixed upstream in
`1ced328b7` (2026-07-24) with regression tests and a doctor check already shipped, so
the downstream rule file documenting it is stale and should be retired rather than
promoted.

## Decisions recorded

| Record | Outcome |
|---|---|
| Agent-neutral host-rules path | `.agents/rules/` (the spelling already reserved in `AUTO_LOADED_RULES_DIR_PREFIXES`); host-authored; managed `AGENTS.md` pointer block; **agy's no-eager-rules exception is not reversed**; wiki softens to query-on-demand |
| Ratchet policy | Absolute floors stay; generic creep removed only where a named deterministic non-regression invariant replaces it, per family, brownfield migration first |
| In-session ticket ready role | Explicit `build_ready: true` for complete in-session defects; omitted normalizes to **not ready** (breaking for GitHub/Linear, intended); `exploratory-qa`'s `ready=false` ratified as the named human-gate exception |
| Credential substrate precedence | Configured provider's token/CLI path first when bootstrap is present and identity-matched; interactive MCP as fallback; generalizes the tenant-safety rationale that already governs Atlassian writes |

## Test plan

Docs and decision records only — no runtime behavior changes. `src/core/upstream-evidence-manifest.ts` is regenerated in the same commit to register the two new tracked files, as the manifest gate requires.

🤖 Generated with Claude Code

## Summary by CodeRabbit

* **Documentation**
  * Added comprehensive improvement notes covering workflows, quality policies, testing, knowledge management, security, UI provenance, and configuration.
  * Added an implementation plan with phased work units, sequencing, acceptance criteria, and recorded decisions.
  * Documented canonical agent rules, credential-provider precedence, ticket-readiness requirements, and ratchet-policy guidance.
  * Added four decision records and linked them from the wiki index.
* **Chores**
  * Updated the evidence manifest to include the new planning and decision documents.

### CODE REVIEW

File: docs/wiki-inbox/2026-08-12-lisa-improvement-notes.md [hunk 1]
```diff
@@ -0,0 +1,450 @@
...
+| L | Fleet BDD gate resync: migrate the three in-scope v1-monolith adopters (tunnlai, geminisportsai, propswap frontend; admin-frontend excluded by owner call) onto Lisa's v2 gate with v1→v2 map/waiver migration, CI enforcement, and upstream-harvest of the downstream inventions (discovery/exclusions, flake classifier, zero-coverage-arm detection) | Host fleet + Lisa BDD gate | G |
+| M | Lisa bug burn-down from the 2026-08-12 fleet audit: skip_jobs substring matching, create-only guard delivery, learnings-ledger-in-rules-dir data loss, ledger trim/whitelist, sync fillMissing provenance, Linear ready→Todo, never-succeeding claude.yml callers, silent postinstall, worktree corruption | Lisa | — |
+| N | Absorb fleet inventions: false-green/zero-coverage family, loop standing-rulings memory, agent-artifact PII scan, deploy-verification timing, claim-time already-implemented guard, retry-by-side-effect-class, pre-commit extension slot | Lisa | — |
+
+A–D, G, H, and J require Lisa changes. E, I, and most of F are downstream. The Lisa-owned portion of item 17 is an existing capability to verify; item 3 is mostly existing capability plus the enforcement-tightening decision in E.
```
[the remark this answers is not in the document]
Comment (CodySwannGT) [reply docs/wiki-inbox/2026-08-12-lisa-improvement-notes.md:439]: Partially accepted in 91ebc235f.

**Accepted — the plan header.** It said `A–J` while the document defines work units through N, which is exactly the dispatch hazard you describe. The header now reads `A–N` and names what each of the four added units owns (K fleet rules cleanup, L fleet BDD gate resync, M Lisa bug burn-down, N absorb fleet inventions), with a note that A–J come from the source document's table and K–N were added by the fleet survey.

**Declined — editing the source note.** `docs/wiki-inbox/2026-08-12-lisa-improvement-notes.md` is an ingestion *source* document, not a planning surface. Per `.claude/rules/documentation-source-paths.md` the inbox holds reader-safe source evidence, and its "Suggested work units" table is a record of what the working session actually proposed. K–N did not exist at that point; they were derived later from the fleet survey. Back-filling them into the source would make the note claim a provenance it does not have and would break the plan header's own "A–J match the source document" statement. The consistency you are asking for is real, and the plan is the correct place to carry it.

File: plans/improvement-notes-implementation.md [hunk 2]
```diff
@@ -0,0 +1,631 @@
+# Implementation plan: 2026-08-12 improvement notes
+
+> Source: `docs/wiki-inbox/2026-08-12-lisa-improvement-notes.md` (revised 2026-08-12 —
+> every "current state" claim in it was verified against the codebase; this plan builds
+> on the corrected version).
+> Status: proposed — Phase 0 decisions must be recorded before Phase 2 starts.
```
[the remark this answers is not in the document]
Comment (CodySwannGT) [reply plans/improvement-notes-implementation.md]: Accepted and fixed in 91ebc235f. Good catch — this was a live gate, not just stale prose: an orchestrator reading line 6 literally would have refused to dispatch Phase 2 work that is already unblocked.

The header now reads `proposed, **Phase 0 complete**` and states that all six decisions were recorded on 2026-08-12 so the Phase 2 gate is satisfied, while keeping the implementation plan itself proposed until each work unit ships — which is the distinction you suggested preserving.

File: plans/improvement-notes-implementation.md [hunk 3]
```diff
@@ -0,0 +1,631 @@
...
+
+### WU-L — fleet BDD gate resync *(follows G part 1; fleet-side)*
+
+**Scope.** No downstream adopter runs Lisa's v2 gate, and **no repo runs the gate in
+CI at all** — it went red on gemini's default branch twice in four days unnoticed;
+tunnlai wired it as a jest arm inside Quality Checks (the only enforcement in the
+fleet); propswap runs it by hand. Fixes are hand-carried between forks (the
+wrapped-tag parser bug was fixed independently in gunnertech `53f49f5`, tunnlai
+`efbafd8a`, gemini `b4ff482e1`, propswap `af4bad9a` + `e0106be`). This work unit ends
+the fork family.
```
[the remark this answers is not in the document]
Comment (CodySwannGT) [reply plans/improvement-notes-implementation.md]: Accepted and fixed in 91ebc235f. You read it correctly: the paragraph collapsed two different gates into one word and so contradicted itself within three lines.

The intended claim was about Lisa's **v2** gate; tunnlai's Jest arm runs its own forked **v1** monolith. Rewritten to say so explicitly — no adopter runs v2, no repo runs *v2* in CI at all, and the only CI enforcement anywhere in the fleet is tunnlai's v1 fork wired into Quality Checks (propswap runs its v1 fork by hand). The fleet baseline is now stated accurately, which matters because WU-L's whole purpose is ending that fork family.

File: plans/improvement-notes-implementation.md [hunk 4]
```diff
@@ -0,0 +1,631 @@
...
+   (today MCP-first) to cite the contract; `lisa-atlassian-access` reads move to
+   token-first with acli as identity-matched fallback (writes already comply).
+3. `lisa-secrets-access` stays the chokepoint feeding the token path; the `tool:` note
+   line already declares which CLIs a credential drives.
+4. Verify live: a Bitwarden-configured project with the bootstrap token present
+   resolves Linear/Notion/Atlassian operations without touching browser auth; pulling
+   the token restores MCP fallback.
+
+**Done when.** Every `*-access` skill resolves substrates in the contract's order, and
+both live checks in step 4 pass.
```
[the remark this answers is not in the document]
Comment (CodySwannGT) [reply plans/improvement-notes-implementation.md]: Accepted and fixed in 91ebc235f — and on checking, the gap was **wider than you reported**, so thank you for pushing on it.

The done-when said "every `*-access` skill" while the steps touched three. Auditing all eight showed three more are MCP-first with the identical headless failure mode:

- `lisa-sentry-access` — MCP → `sentry-cli` → `SENTRY_AUTH_TOKEN` REST
- `lisa-posthog-access` — MCP → `POSTHOG_PERSONAL_API_KEY`
- `lisa-jam-access` — MCP → `JAM_PAT` CLI

So **six** skills bind the contract, not three, and those three are now added to step 2's re-ordering work rather than being quietly covered by a done-when nobody would have traced.

On your two named candidates, both are genuine exceptions but for different reasons, and both are now recorded in the plan as auditable exclusions rather than omissions:

- `lisa-secrets-access` — it *is* the token source feeding every other skill's tier 1, so it has no substrate tier of its own to order.
- `lisa-sonarcloud-access` — single substrate (the official SonarQube MCP server, already authenticated headlessly from `SONARQUBE_CLI_TOKEN`), so there is no second tier to prefer and no browser-auth failure mode to fix.

Each of the two will carry a one-line "out of scope because …" note in its own text, per your suggestion.

File: plans/improvement-notes-implementation.md [hunk 5]
```diff
@@ -0,0 +1,631 @@
...
+### WU-M — Lisa bug burn-down from the fleet audit *(Phase 1; decision-free)*
+
+File one tracked ticket per row of the bugs table (checking first which are already
+filed — #2395 is; the guard-delivery and skip_jobs bugs may overlap existing issues),
+fix in severity order, and where a downstream repo built a compensating guard
+(`check-ledger-archive`, `check-lifecycle-label-config`, `check-skipped-required-checks`,
+`check-lisa-pii`), evaluate absorbing the guard upstream as the regression test for
+the fix. The three High rows are candidates to pull ahead of everything else in this
+plan. **Done when** every row is fixed-or-ticketed, and the fixed ones reach installed
+repos — which itself depends on fixing the guard-delivery row.
```
[the remark this answers is not in the document]
Comment (CodySwannGT) [reply plans/improvement-notes-implementation.md:646]: Accepted and fixed in 91ebc235f. The bug table had been updated to mark the row refuted but WU-M's instructions had not caught up, so the section still told an implementer to file a ticket for it and to pull "the three High rows" ahead of everything else.

WU-M now opens by stating that the `skip_jobs` row is refuted and closed — fixed upstream in `1ced328b7` (2026-07-24, #2028), gets no fix ticket, and must not be promoted downstream, with WU-K retiring the stale downstream rule file that still documents it. Its two residual gaps stay tracked separately as #2426 and #2427. The count is corrected to **two** surviving High rows, and the done-when now admits "recorded as refuted with the upstream commit that already fixed it" as a valid terminal state alongside fixed-or-ticketed.

That last part generalizes your finding: this row will not be the only one, so the section now leads with "verify every row against current source before fixing it, and ticket only the rows that survive."

File: wiki/decisions/2026-08-12-agent-neutral-host-rules-path.md [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+
+### 1. The canonical host-rules path is `.agents/rules/`
+
+Host-authored, agent-neutral, one directory. Chosen over `agents/rules/` because the
+dot-prefixed spelling matches the existing `.agents/` tree (which already holds
+`skills/` and `plugins/`) and because `.agents/rules` is **already reserved** in
+`AUTO_LOADED_RULES_DIR_PREFIXES` (`src/core/project-config.ts:67`) — the blocklist
+that keeps the learnings ledger from resolving into an eager-rule tree. Adopting the
+spelling the guard already anticipates avoids a rename later and inherits the
+ledger-collision protection for free.
```
[the remark this answers is not in the document]
Comment (CodySwannGT) [reply wiki/decisions/2026-08-12-agent-neutral-host-rules-path.md:57]: Partially accepted in 91ebc235f — the substance, not the rename.

**Accepted.** You are right that choosing a canonical path does not confer universal auto-loading, and the record did not say so anywhere. It now does, in item 1: `.agents/rules/` is a canonical *source* path, only Antigravity discovers it natively, and the per-agent reality is spelled out (Claude Code `.claude/rules/`, Cursor `.cursor/rules/*.mdc`, Copilot `.github/copilot-instructions.md` plus `.github/instructions/*.instructions.md`, Codex's root-down `AGENTS.md` chain). The text now states explicitly that the pointer block makes the directory *reachable* on agents that read `AGENTS.md` and follow relative references, and does **not** make it eagerly loaded anywhere except Antigravity. Work unit A owns the per-agent bridge and must state, per agent, whether rules arrive eagerly or on demand — with the parity tests asserting that real behavior rather than uniform delivery.

**Declined — the rename.** "Canonical host-rules source" is what the record now says in prose, so the title would be redundant, and the heading is cited by name from the implementation plan and from the D1 row of the Phase 0 table. Renaming a decision record's heading to fix a claim the body can state directly costs more than it buys.

One correction to your framing: Antigravity's no-eager-rules gap (item 5) is not an instance of this problem — it is the opposite case. That gap is about agy's plugin hooks not firing in headless mode, and it is accepted rather than bridged. The record now cross-references the two so they are not read as the same issue.

File: wiki/decisions/2026-08-12-agent-neutral-host-rules-path.md [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+
+### 2. Ownership is split and explicit
+
+- `.agents/rules/` is **host-authored**. Lisa never writes rule bodies into it.
+- Lisa's own shared rules keep originating at `plugins/src/base/rules/` and reaching
+  agents through the existing per-agent build fan-out and `inject-rules.sh`.
+- `AGENTS.md` receives a **Lisa-managed pointer block** naming the host-rules
+  directory. Pointer only — never rule bodies. This is the mechanism the 2026-06-06
+  update established (canonical, rule-free `AGENTS.md`) and it is preserved.
```
[the remark this answers is not in the document]
Comment (CodySwannGT) [reply wiki/decisions/2026-08-12-agent-neutral-host-rules-path.md:73]: Accepted and fixed in 91ebc235f. This was a real contradiction, not a wording problem — as written, one line forbade exactly what another line scheduled.

Resolved in favour of the ownership rule, by defining what WU-B actually does: B does **not** ship a Lisa-generated pack into the host directory. `.agents/rules/` is the pack's **destination rung**, and content arrives there by human-gated promotion through the ladder — the gardener proposes, a human approves the ticket, and the promoted rule is written as host content the host then owns. Lisa's own shared rules never take that route; they stay on the `plugins/src/base/rules/` fan-out.

So, to your three questions directly: the **producer** is the gardener proposing plus a human approving, the **destination** is `.agents/rules/`, and the **generated-file policy** is that there is no generated-file class there at all — which means nothing for apply to overwrite or reconcile, and the create-only posture in item 4 stays coherent. Both line 49 and the Consequences entry now say this.

File: wiki/decisions/2026-08-12-agent-neutral-host-rules-path.md [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+
+### 4. Migration for installed projects is an idempotent managed block
+
+`AGENTS.md` stays create-only for **overwrites**, but apply/doctor already makes
+bounded managed edits to an existing file (`src/core/instruction-files-migration.ts`
+strips the legacy `LISA_RULES_START..END` block and reconciles the agy
+project-learnings bridge). The pointer block extends that existing machinery: add or
+update the block, never touch surrounding host prose, no-op on repeat runs. A
+template-only change is insufficient — current apply intentionally does not overwrite
+an existing `AGENTS.md`.
```
[the remark this answers is not in the document]
Comment (CodySwannGT) [reply wiki/decisions/2026-08-12-agent-neutral-host-rules-path.md:94]: Declining this one as scoped, with reasoning — please reopen if you disagree.

This is a **decision record**, and the decision it was asked to make was *whether* `AGENTS.md` carries a Lisa-managed pointer block and *by what mechanism*. Item 4 answers both: it extends the existing bounded-edit machinery in `src/core/instruction-files-migration.ts` — the same code that already strips the legacy `LISA_RULES_START..END` block and reconciles the agy project-learnings bridge — and it states the invariants that machinery must satisfy (add or update the block, never touch surrounding host prose, no-op on repeat runs), plus why a template-only change is insufficient.

What you are asking for — an exact marker pair, exact generated text, placement, and a test matrix covering malformed markers, duplicate markers, legacy block migration, and a missing `.agents/rules/` directory — is an implementation specification. Putting it in the decision record has two concrete costs: it duplicates a contract that will live next to the code, and it freezes literal generated text in a wiki page that no test can hold to account, so the two drift the moment the implementation adjusts a marker. Lisa already has a home for this: work unit A, which owns the migration and whose done-when includes the parity tests.

Your edge-case list is genuinely the right list, though, and it should not be lost between here and A. The legacy `LISA_RULES_START..END` case in particular is a live migration path with existing code, not a hypothetical. I have left it as the enumerated obligation on WU-A rather than dropping it — if you would like it recorded as an explicit checklist on the work unit rather than in this thread, that is a reasonable ask and I will add it there.

File: wiki/decisions/2026-08-12-agent-neutral-host-rules-path.md [hunk 9]
```diff
@@ -0,0 +1,127 @@
...
+
+The eager `wiki-knowledge-source` rule currently says *consult the wiki first* and
+*use what the wiki says as the authoritative answer*. That is stronger and more
+context-expensive than intended. It becomes: the wiki exists, query it when you need
+depth, do not load it at session start. The query skill remains the retrieval path.
+
+A packaging bug is fixed in the same change: the eager rule ships in the **base**
+plugin while `lisa-wiki-query` ships only in the **wiki** plugin, so base-only
+projects get a rule pointing at a skill they do not have. Either condition the rule on
+the wiki plugin's presence or move the skill into base.
```
[the remark this answers is not in the document]
Comment (CodySwannGT) [reply wiki/decisions/2026-08-12-agent-neutral-host-rules-path.md]: Accepted and fixed in 91ebc235f. You identified the contradiction precisely: the paragraph asserted the bug "is fixed in the same change" and then left the remedy unchosen, in a documentation-only PR that fixes nothing.

Rewritten as planned work rather than as a claim of completion. The record now says the packaging bug is **identified here and fixed by work unit A**, keeps both remedies open (condition the eager rule on the wiki plugin's presence, or move `lisa-wiki-query` into base), and states why the choice is delegated rather than settled here — it turns on packaging details A has to work out anyway, specifically whether base may carry a wiki-shaped skill at all.

I also took your last sentence literally and made the consequence explicit rather than leaving it implied: the record now states that until A lands, base-only projects still receive `wiki-knowledge-source` without `lisa-wiki-query`, and that A is not done until that is closed with a test proving a base-only install has no dangling rule. That converts your warning into a done-when someone can fail.

File: wiki/decisions/2026-08-12-credential-substrate-precedence.md [hunk 10]
```diff
@@ -0,0 +1,101 @@
...
+
+## Decision
+
+**One shared, vendor-neutral precedence contract, cited by every `*-access` skill:**
+
+1. **Configured-provider substrate first** — the token or CLI path fed by
+   `lisa-secrets-access` — whenever its bootstrap credential is available *and* the
+   resolved substrate identity-matches the configured tenant/workspace/site.
+2. **Interactive MCP as fallback** — used when the provider path is genuinely
+   unavailable (no bootstrap, no adapter for the operation, provider outage).
```
[the remark this answers is not in the document]
Comment (CodySwannGT) [reply wiki/decisions/2026-08-12-credential-substrate-precedence.md:41]: Accepted, and this is the most valuable finding in the review — fixed in 91ebc235f.

The contract said "provider outage" triggers MCP fallback without distinguishing *when* in the request lifecycle the outage happened, which as you say silently authorizes retrying a write whose side effect may already have landed. Generalizing the Atlassian write rule into a shared contract would have dropped exactly the safeguard the local note existed to provide.

The contract now bounds fallback on mutating operations by whether the side effect can already have occurred:

- **Permitted only when the failure is provably pre-dispatch** — no bootstrap credential, identity mismatch, no adapter for the operation, request rejected before transmission, connection refused, DNS failure, or an authoritative 4xx the provider defines as a no-op rejection.
- **Forbidden after an in-flight write of unknown outcome** — timeout, 5xx, mid-request connection reset, or any response the client cannot classify. These must first **reconcile by reading back** on the same substrate tier: re-read by idempotency key or natural key and decide applied-or-not from observed state. Only once reconciliation proves the write did not land may it be retried, on either substrate.
- **When reconciliation is impossible** (the read path is down too), fail closed with an operator-readable "write outcome unknown — verify before retrying" rather than silently retrying anywhere.

Reads keep free access to tier 2. The record states that this preserves rather than loosens the Atlassian safeguards — their tenant-binding rule stays and this adds the outcome-unknown rule on top — and routes the per-vendor reconciliation mechanics to work unit J, with the contract stating only the boundary.

File: wiki/decisions/2026-08-12-in-session-ticket-ready-role.md [hunk 11]
```diff
@@ -0,0 +1,113 @@
...
+defect it found next to it, and then left the new ticket sitting. Filing without the
+ready role is an incomplete handoff — no other agent will ever pick it up.
+
+Current behavior is inconsistent in three distinct ways.
+
+**1. Loops differ, some deliberately.** `lisa-monitor`, `lisa-verify-prd`, and
+`lisa-repair-intake` file with `build_ready: true`. `lisa-exploratory-qa` deliberately
+defaults `ready=false`, creating findings in the backlog for a human to review and
+promote. The standard in-session `lisa-track` path creates complete unmatched work as
+a build-ready leaf before claiming it.
```
[the remark this answers is not in the document]
Comment (CodySwannGT) [reply wiki/decisions/2026-08-12-in-session-ticket-ready-role.md:22]: Accepted and fixed in 91ebc235f. Item 3 required "an explicit marker" and then never named it, which for an Accepted decision is an underspecification an implementer would have had to guess at.

**The marker is `build_ready: false` passed explicitly. There is no new field.** Once item 2 makes omission mean not-ready, the existing write-control input carries three states instead of two, and the third is the marker. The record now carries the table:

| `build_ready` | Lifecycle outcome | Handoff status |
|---|---|---|
| `true` | ready role applied | complete — build-intake claims it |
| `false` (explicit) | no ready role | complete — a human product call is pending |
| omitted | no ready role | **incomplete** — writers warn |

This is also what makes item 4 coherent, and it resolves your second point. You are right that `ready=false` is an exploratory-QA parameter rather than a persisted marker — but under this contract that is the design, not a defect: the distinction is caller-supplied and lives in the call, so what persists is simply the absence of the ready role, and the marker's job is to certify **at write time** that the absence was chosen rather than defaulted. Exploratory QA is therefore not an exception to the marker rule; its `ready=false` *is* the marker.

Scope is stated as a `lisa-tracker-write` contract obligation binding all three vendor writers identically, implemented by work unit D alongside item 2. Omission warns rather than hard-fails so a legacy caller degrades to the safe direction instead of breaking. On the writer contracts and `build-ready-control.test.ts` still encoding today's provider behavior — correct, and that is the change WU-D makes; see the reply on the companion thread for the enumerated surfaces and test matrix.

File: wiki/decisions/2026-08-12-in-session-ticket-ready-role.md [hunk 12]
```diff
@@ -0,0 +1,113 @@
...
+All three vendor writers converge on: omission means the item does **not** enter the
+ready role. Ready becomes an explicit claim, never an accident of which tracker a
+project uses.
+
+This changes GitHub and Linear behavior, so it is a breaking change for any caller
+relying on the old implicit-ready default. That is acceptable and in fact the point:
+every Lisa call site is being made explicit anyway (item 1 above), so the only paths
+affected are ones that were silently depending on a provider-specific default — which
+is exactly the class of bug this decision exists to eliminate. GitHub's validator
+normalization (omitted → true) is removed in the same change.
```
[the remark this answers is not in the document]
Comment (CodySwannGT) [reply wiki/decisions/2026-08-12-in-session-ticket-ready-role.md:68]: Both comments accepted, fixed in 91ebc235f. The second one caught a factual error worth calling out.

**On `LINEAR_WORKFLOW_DEFAULTS.ready` — you are right, and the record was wrong.** It is `"Ready"`, not `"Todo"` (`src/sync/lifecycle-defaults.ts:65-71`). tunnlai `a8899a17` found a real bug, but it was **already fixed upstream** before this record was written, so item 5 was scheduling a WU-M fix for something that did not need fixing. Context 3 now records the verification and the correction, and item 5 is rewritten: the *invariant* is ratified (the ready state must be a lane a human moves an issue into, never the default the tracker assigns on creation), but WU-M carries **no Linear adapter fix**. What survives is the narrower configuration guard — `lisa-validate-tracker-mapping` should fail when a project's configured `ready` state is one the tracker assigns on creation, which is the form the bug can still take in the field.

This is the second row in this PR where downstream evidence lagged current Lisa (the `skip_jobs` row was the first), so the record links it to that meta-finding rather than treating it as a one-off.

**On separating decision status from implementation status** — accepted. Consequences now leads with it explicitly: nothing in the record describes shipped behavior except item 5's adapter fix, which was already in. WU-D implements items 1–3 and 6; WU-M carries only item 5's configuration guard.

**On the writer contracts and validator** — accepted, and enumerated so the work is traceable rather than implied. The record now names the current, unchanged surfaces: `lisa-linear-write-issue` creates a leaf in the resolved `ready` state by default, `lisa-github-write-issue` applies `status:ready` unless `n: false`, and `lisa-github-validate-issue` explicitly normalizes an omitted `n` to `true`. All three flip to not-ready-on-omission and that validator normalization is removed. Test coverage is specified as your four cases per vendor — omitted, explicit `false`, explicit `true`, and exploratory QA's explicit `ready=false` — in **both** the skill sources and the generated per-agent plugin artifacts, since the artifacts are what agents actually load.
